notebooks/scrf.ipynb

```diff
@@ -0,0 +1,828 @@
+    "    \"pr\": 404.29331338341586,\n",
+    "}\n",
+    "\n",
+    "# in units of day\n",
+    "temporal_scales = {\n",
+    "    \"tasmax\": 3.725572253986237,\n",
+    "    \"tasmin\": 3.9400152223556937,\n",
+    "    \"pr\": 2.0457947933307676,\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f6204cf5-4d1c-4641-8363-1eedc6310722",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import dask\n",
+    "\n",
+    "\n",
+    "def make_scrf_path(\n",
+    "    obs: str,\n",
+    "    label: str,\n",
+    "    start_year: str,\n",
+    "    end_year: str,\n",
+    "    **kwargs,\n",
+    "):\n",
+    "    return f\"scrf/{obs}_{label}_{start_year}_{end_year}.zarr\"\n",
+    "    # return f\"scrf/{crs.split('=')[-1]}_test.zarr\"\n",
+    "\n",
+    "\n",
+    "# @dask.delayed\n",
+    "def generate_scrf(\n",
+    "    variable,\n",
+    "    spatial_scales,\n",
+    "    temporal_scales,\n",
+    "    start_year,\n",
+    "    end_year,\n",
+    "):\n",
+    "    if variable == \"temp\":\n",
+    "        path = make_scrf_path(\"ERA5\", \"tasmin\", start_year, end_year)\n",
+    "        store = fsspec.get_mapper(f\"az://flow-outputs/intermediate/\" + path)\n",
+    "    else:\n",
+    "        path = make_scrf_path(\"ERA5\", \"pr\", start_year, end_year)\n",
+    "        store = fsspec.get_mapper(f\"az://flow-outputs/intermediate/\" + path)\n",
+    "\n",
+    "    print(path)\n",
+    "    if \".zmetadata\" in store:\n",
+    "        return (\"skipping\", path)\n",
+    "\n",
+    "    # get the correct temporal and spatial scales to use\n",
+    "    print(\"initializing model\")\n",
+    "    if variable == \"temp\":\n",
+    "        ss = np.round(\n",
+    "            np.mean([spatial_scales[\"tasmax\"], spatial_scales[\"tasmin\"]]), 1\n",
+    "        )\n",
+    "        ts = np.round(\n",
+    "            np.mean([temporal_scales[\"tasmax\"], temporal_scales[\"tasmin\"]]), 1\n",
+    "        )\n",
+    "        v = \"tasmax\"\n",
+    "    elif variable == \"pr\":\n",
+    "        ss = np.round(spatial_scales[\"pr\"], 1)\n",
+    "        ts = np.round(temporal_scales[\"pr\"], 1)\n",
+    "        v = \"pr\"\n",
+    "\n",
+    "    # initializing model\n",
+    "    model = gs.Gaussian(dim=3, var=1.0, len_scale=[ss, ss, ts])\n",
+    "    srf = gs.SRF(model, seed=int(start_year) + len(variable))\n",
+    "\n",
+    "    print(\"getting template\")\n",
+    "    template = get_obs(\n",
+    "        obs=\"ERA5\",\n",
+    "        train_period_start=1980,\n",
+    "        train_period_end=1980,\n",
+    "        variables=v,\n",
+    "        chunking_approach=None,\n",
+    "    )[v].isel(time=0)\n",
+    "    template[\"lon\"] = convert_long3_to_long1(template.lon)\n",
+    "    template = template.reindex(lon=sorted(template.lon.values))\n",
+    "    template = template.rename({\"lon\": \"x\", \"lat\": \"y\"})\n",
+    "\n",
+    "    mill_template = xr.DataArray(\n",
```

Comment:
a small comment here about this being the point where the miller projection comes into play

Reply:
added